### DISCUSSION TITLE
Generate OpenAPI Document in YAML Format

### DISCUSSION THREAD

**Discussion by w1am** [dotnet/aspnetcore#58234] [Q&A] [answered]
I am using the .NET 9 RC version of [Microsoft.AspNetCore.OpenApi](https://www.nuget.org/packages/Microsoft.AspNetCore.OpenApi/9.0.0-rc.1.24452.1).

The default behavior of [MapOpenApi](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.openapiendpointroutebuilderextensions.mapopenapi?view=aspnetcore-9.0&viewFallbackFrom=net-9.0) seems to generate the OpenAPI document in JSON format. After reviewing the documentation, I couldn't find a way to configure the format to YAML.

Is this possible?

**Comment by martincostello** [comment, marked as answer]
There's no built-in ability to generate YAML for OpenAPI in .NET 9.

I've added the ability to do so to a library I've been working on if that's useful to you: https://github.com/martincostello/openapi-extensions

**Reply by rajashrimense** [reply]
Hi, where you updated this YAML format. Actually, my issue is in my workflow instead of **format="openapi+json"** i want to add **format="openapi+yaml"**

**Reply by martincostello** [reply]
I don't fully understand your question, but the code that sets the `Content-Type` header to signal the format is YAML is [here](https://github.com/martincostello/openapi-extensions/blob/cbb384770b6a235e11681ab72fb31aae65074dab/src/OpenApi.Extensions/OpenApiEndpointRouteBuilderExtensions.cs#L61).
